datalad/support/annexrepo.py

```diff
@@ -597,12 +597,38 @@ def get_size_from_key(key):
         """A little helper to obtain size encoded in a key"""
         if not key:
             return None
-        try:
-            size_str = key.split('-', 2)[1].lstrip('s')
-        except IndexError:
-            # has no 2nd field in the key
+
+        # TODO: this method can be more compact. we don't need particularly
+        #       elaborated error distinction
```

Comment:
I'm not sure what this is referring to.  I spot only one exception that's raised.

Reply:
TBH, I don't remember. Might be a leftover from an earlier version. Will have a closer look when I write the tests for it and delete, if I can't decode that comment again (otherwise clarify or resolve it, of course) ;-)

Reply:
See below. @yarikoptic kind of found what this was about. ;-)